idaes/power_generation/unit_models/drum1D.py

```diff
@@ -0,0 +1,1171 @@
+        default=1.0,
+        description='inside diameter of drum',
+        doc='define inside diameter of drum'))
+
+    CONFIG.declare("drum_thickness", ConfigValue(
+        default=0.1,
+        description='drum wall thickness',
+        doc='define drum wall thickness'))
+
+    def build(self):
+        """
+        Begin building model
+        """
+
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        # Build Control Volume
+        self.control_volume = ControlVolume0DBlock(default={
+                "dynamic": self.config.dynamic,
+                "has_holdup": self.config.has_holdup,
+                "property_package": self.config.property_package,
+                "property_package_args": self.config.property_package_args})
+
+        self.control_volume.add_geometry()
+
+        self.control_volume.add_state_blocks(has_phase_equilibrium=False)
+
+        self.control_volume.add_material_balances(
+            balance_type=self.config.material_balance_type,)
+
+        self.control_volume.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_transfer=self.config.has_heat_transfer)
+
+        self.control_volume.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=True)
+
+        self.flash = HelmPhaseSeparator(
+            default={
+                "dynamic": False,
+                "property_package": self.config.property_package,
+            }
+        )
+
+        self.mixer = HelmMixer(
+            default={
+                "dynamic": False,
+                "property_package": self.config.property_package,
+                "momentum_mixing_type": MomentumMixingType.equality,
+                "inlet_list": ["FeedWater", "SaturatedWater"]
+                }
+            )
+
+        # Inlet Ports
+        # FeedWater to Drum (from Pipe or Economizer)
+        self.feedwater_inlet = Port(extends=self.mixer.FeedWater)
+        # Sat water from water wall
+        self.water_steam_inlet = Port(extends=self.flash.inlet)
+
+        # Exit Ports
+        # Liquid to Downcomer
+        # self.liquid_outlet = Port(extends=self.mixer.outlet)
+        self.add_outlet_port('liquid_outlet', self.control_volume)
+        # Steam to superheaters
+        self.steam_outlet = Port(extends=self.flash.vap_outlet)
+
+        self.stream_flash_out = Arc(
+            source=self.flash.liq_outlet, destination=self.mixer.SaturatedWater
+            )
+
+        # Pyomo arc connect flash liq_outlet with mixer SaturatedWater inlet
+        TransformationFactory("network.expand_arcs").apply_to(self)
+
+        # connect internal units (Mixer to Water Tank Model)
+        # Mixer Outlet (mixed_state) to unit control volume.properties_in
+        @self.Constraint(self.flowsheet().config.time)
+        def connection_material_balance(b, t):
+            return 1e-4*b.mixer.mixed_state[t].flow_mol == \
```

Comment:
Rather than adding fixed scaling factors here, we really should use the scaling tools.